### PR TITLE
feat(frontend): add Jest unit testing infrastructure

### LINKED ISSUE #6442: feat(frontend): Add Jest unit testing infrastructure
## Summary

Add Jest unit testing infrastructure to the frontend codebase to enable fast, isolated unit tests alongside the existing TestCafe E2E tests.

## Motivation

The frontend currently relies on TestCafe for E2E testing, which is excellent for integration tests but can be slow and complex for testing individual utility functions and business logic. Adding Jest provides:

- **Fast feedback loop** - Unit tests run in milliseconds, not minutes
- **Isolated testing** - Test pure functions and utilities without browser overhead
- **Better coverage** - Easier to test edge cases and error conditions
- **CI integration** - Quick unit test checks on every PR

### Jest Configuration
- Add Jest with `ts-jest` preset for TypeScript support
- Configure module path aliases (`common/`, `components/`, `project/`)
- Set up coverage collection for `common/` and `web/` directories
- Support for both `__tests__/` directory pattern and `*.test.ts` files

### CI/CD Integration
- Add new GitHub Actions workflow (`frontend-pull-request.yml`)
- Runs unit tests automatically on PRs that modify frontend code
- Uses Node.js version from `.nvmrc` for consistency

### Initial Test Coverage
- Add comprehensive unit tests for `featureFilterParams` utility (245 lines)
  - Tests for `buildUrlParams`, `buildApiFilterParams`, `getFiltersFromParams`, `hasActiveFilters`
  - Covers archived feature filtering logic

### Dependencies Added
- `jest` - Test runner
- `ts-jest` - TypeScript preprocessor for Jest
- `@types/jest` - TypeScript definitions

## Testing

```bash
cd frontend
npm run test:unit
```

## Related

- Spike: #6426

---

Note: This branch also includes fixes for versioned environment toggle functionality and E2E test improvements, which may be split into separate PRs if needed.

### PR DESCRIPTION
**Pull request by talissoncosta**
## Summary

Add Jest unit testing infrastructure to the frontend codebase to enable fast, isolated unit tests alongside the existing TestCafe E2E tests.

Closes #6442

### Jest Configuration
- Add Jest with `ts-jest` preset for TypeScript support
- Configure module path aliases (`common/`, `components/`, `project/`)
- Set up coverage collection for `common/` and `web/` directories
- Add `tsconfig.jest.json` with `jsx: react` for parsing TSX files in tests

### CI/CD Integration
- Add new GitHub Actions workflow (`frontend-pull-request.yml`)
- Runs unit tests automatically on PRs that modify frontend code
- Add unit test job to `frontend-deploy-production.yml`
- Use `--passWithNoTests` flag to prevent failures if tests are accidentally deleted

### Initial Test Coverage
- Add comprehensive unit tests for `featureFilterParams` utility
  - Tests cover `buildUrlParams`, `buildApiFilterParams`, `getFiltersFromParams`, `hasActiveFilters`
- Add comprehensive unit tests for `format` utility
  - Tests cover `shortenNumber`, `nearestTen`, `camelCase`, `enumeration`, `fullName`, `truncateText`, `trimAndHighlightSpaces`, `userDisplayName`
- Migrate `format.js` to TypeScript (`format.ts`) with proper type annotations

### Dead Code Removal
- Remove 12 unused functions from `format.ts`: `removeAccents`, `countdown`, `ago`, `initialAndLastName`, `ordinal`, `money`, `cssImage`, `objectEqual`, `countDp`, `age`, `bytesToMB`, `bytesToSize`
- These functions were never imported anywhere in the codebase

### Test Style
- Use Jest's `it.each` with tagged template literals for table-driven tests
- Concise, readable test cases with clear input/output pairs

### Dependencies Added
- `jest`, `ts-jest`, `@types/jest`

## Known Issues (Out of Scope)

These are pre-existing behaviours documented by tests, not introduced by this PR:

- `fullName({ firstName: 'kyle' })` returns `'Kyle '` with trailing space - existing behaviour, fix in future PR

## How did you test this code?

```bash
cd frontend
npm run test:unit
```

All 71 tests pass.

## Checklist

- [ ] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below
- [x] I have filled in the "How did you test this code" section below
- [x] I have used a Conventional Commit title for this Pull Request

### CODE REVIEW

File: .github/workflows/frontend-pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Frontend Pull Requests
```
Comment (Zaimwa9) [inline .github/workflows/frontend-pull-request.yml:1]: I like what i'm seeing there 👍

Comment (talissoncosta) [reply .github/workflows/frontend-pull-request.yml:1]: I think it worth to have a unit test structure, unit tests does not cost too much to implement and are quick to run.

Comment (Zaimwa9) [inline .github/workflows/frontend-pull-request.yml:1]: Let's also add the tests in the deploy to production workflow

File: frontend/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/** @type {import('jest').Config} */
+module.exports = {
+  collectCoverageFrom: [
```
Comment (Zaimwa9) [inline frontend/jest.config.js:4]: I remember a clearMocks prop maybe that could be worth adding

**Review by Zaimwa9 — COMMENTED** [review]
Just adding a prop and to the deploy to production pipeline and should be good to go. Thanks for the initiative!

File: frontend/common/utils/format.ts [hunk 3]
```diff
@@ -116,81 +127,87 @@ const Format = {
...
+  removeAccents(str: string | null | undefined): string | null | undefined {
     // Sergio Agüero > Sergio Aguero
     if (!str) {
       return str
     }
 
+    let result = str
     for (let i = 0; i < Utils.accents.length; i++) {
-      str = str.replace(Utils.accents[i].letters, Utils.accents[i].base)
+      result = result.replace(Utils.accents[i].letters, Utils.accents[i].base)
```
Comment (Zaimwa9) [inline frontend/common/utils/format.ts]: Just to confirm Utils is global and not breaking here ?

**Review by Zaimwa9 — COMMENTED** [review]
One comment and ready to be approved

File: frontend/common/utils/format.ts (hunk 3 above)
Comment (talissoncosta) [reply frontend/common/utils/format.ts]: Thanks for raising this. Utils actually works pretty well but this change was actually not needed. Reverting it.

**Review by Zaimwa9 — DISMISSED** [review]
👍

**Review by Zaimwa9 — DISMISSED** [review]

**Review by Zaimwa9 — APPROVED** [review]
